src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -50,103 +57,228 @@
 			this.target = target;
 		}
+
 		/**
 		 * @return the orgId
 		 */
 		public String getOrgId() {
 			return orgId;
 		}
+
 		/**
 		 * @param orgId the orgId to set
 		 */
 		public void setOrgId(String orgId) {
 			this.orgId = orgId;
 		}
+
 		/**
 		 * @return the spaceId
 		 */
 		public String getSpaceId() {
 			return spaceId;
 		}
+
 		/**
 		 * @param spaceId the spaceId to set
 		 */
 		public void setSpaceId(String spaceId) {
 			this.spaceId = spaceId;
 		}
+
 		/**
 		 * @return the applicationId
 		 */
 		public String getApplicationId() {
 			return applicationId;
 		}
+
 		/**
 		 * @param applicationId the applicationId to set
 		 */
 		public void setApplicationId(String applicationId) {
 			this.applicationId = applicationId;
 		}
+
 		/**
 		 * @return the accessURL
 		 */
 		public String getAccessURL() {
 			return accessURL;
 		}
+
 		/**
 		 * @param accessURL the accessURL to set
 		 */
 		public void setAccessURL(String accessURL) {
 			this.accessURL = accessURL;
 		}
+
+		/**
+		 * @return the isInternal
+		 */
+		public boolean isInternal() {
+			return internal;
+		}
+
+		/**
+		 * @param isInternal the isInternal to set
+		 */
+		public void setInternal(boolean isInternal) {
+			this.internal = isInternal;
+		}
+
 		/**
 		 * @return the numberOfInstances
 		 */
 		public int getNumberOfInstances() {
 			return numberOfInstances;
 		}
+
 		/**
 		 * @param numberOfInstances the numberOfInstances to set
 		 */
 		public void setNumberOfInstances(int numberOfInstances) {
 			this.numberOfInstances = numberOfInstances;
 		}
-		
+
 	}
-	
+
+	private Flux<AppRouteVector> determineRoutes(String orgId, String spaceId) {
+		Mono<ListRouteMappingsResponse> routeMappings = this.cfAccessor.retrieveRouteMappings();
+
+		Flux<RouteMappingResource> something = routeMappings
+				.flatMapMany(mapper -> Flux.fromIterable(mapper.getResources()));
+
+		Flux<AppRouteVector> routeMappingFlux = something.map(mapper -> {
+			AppRouteVector route = new AppRouteVector();
+
+			// TODO some validation here
+
+			route.setAppId(mapper.getEntity().getApplicationId());
+			route.setPort(mapper.getEntity().getApplicationPort());
+			route.setRouteId(mapper.getEntity().getRouteId());
+			return route;
+		});
+
+		Flux<List<RouteResource>> spaceRoutes = routeMappingFlux.flatMapSequential(v -> {
+			return this.cfAccessor.retrieveSpaceRoutes(spaceId).flatMap(mapper -> Mono.just(mapper.getResources()));
+		});
+		Flux<AppRouteVector> routeInstanceFlux = Flux.zip(routeMappingFlux, spaceRoutes).flatMap(tuple -> {
+			AppRouteVector v = tuple.getT1();
+			List<RouteResource> routes = tuple.getT2();
+
+			if (routes.size() == 0) {
+				// NB: This drops the current instance route!
+				return Mono.empty();
+			}
+
+			// find the route to match this InstanceRoute
+			RouteResource route = routes.stream().filter(r -> r.getMetadata().getId().equals(v.getRouteId()))
+					.findFirst().get();
+
+			// TODO what if we dont find???
```

Comment:
yeah, that's what I mean ;-)

Reply:
Yup i need to go back and add in some "robustness" once we are happy with the implementation..